File: Valid_Anagram.py
class Solution:
    def isAnagram(self, s: str, t: str) -> bool:
        if len(s) != len(t):
            return False
        d = {}
        for elm in s:
            if elm in d:
                d[elm] += 1
            else:
                d[elm] = 1
        for elm in t:
            if elm in d:
                if d[elm] == 1:
                    d.pop(elm)
                else:
                    d[elm] -= 1
            else:
                return False
        if len(d) == 0:
            return True
        return False

File: Valid_Anagram.cpp
class Solution {
public:
    bool isAnagram(string s, string t) {
        map<char,int> ss;
        map<char,int> tt;
        if(s.size() !=t.size())
        {
            return false;
        }
        for(int i=0;i<s.size();i++)
        {
            if(ss.find(s[i])==ss.end())
            {
                ss[s[i]] = 1;
            }
            else
            {
                ss[s[i]]+=1;
            }
        }
        for(int j=0;j<t.size();j++)
        {
            if(tt.find(t[j])==tt.end())
            {
                tt[t[j]] = 1;
            }
            else
            {
                tt[t[j]]+=1;
            }
        }
        return tt == ss;
            }
};